src/lib.rs

```diff
@@ -429,8 +429,8 @@ mod internal {
     }
 
     #[doc(hidden)]
-    pub fn precise_time_microsec() -> u64 {
-        time::precise_time_ns() / 1000
+    pub fn precise_time_microsec() -> i128 {
+        (time::OffsetDateTime::now_utc() - time::OffsetDateTime::UNIX_EPOCH).whole_milliseconds()
```

Comment:
it seems like time::OffsetDateTime::now_utc() only is a wraper around std::time::SystemTime so think that we can remove the dependency to time and do 
`SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).expect("SystemTime is before UNIX EPOCH").as_millis as u64` as this is almost the same code but one dependency less I think I prefer this.
casted to u64 as it is long enough and it keeps the size of the event to the old size

Reply:
I made the change you suggested with one adjustment: since `Duration::as_millis()` returns an `u128` the conversion to `u64` can also overflow, so I added another `expect` that the range is not too big. IMO this is still not ideal for a library crate which might be used in scenarios with weird clock gymnastics (e.g., in tests). I think the only solution to this is to use large enough types for time stamps and durations (at which point it might make sense to just directly switch to `SystemTime` and `Duration` for `ts` and `dur`).